GeneralJS/balanced-parens-2.ts

```diff
@@ -16,6 +16,91 @@
  *
  */
 
-function balancedParens2(input): boolean {}
+// solve parens 1 first lol
+
+function balancedParens2(input) {
+    /*
+        ((())[]{}) = true
+        var hubble = function() { telescopes.awesome();) = false
+    */
+   
+    // breaking case : closed symbol is proceeded by a none matching open symbol
+
+    // define regex
+    const parsingRegex = /[\[\]\{\}\(\)]/g;
+    const openRegex = /[\[\{\(]/g
+    const closedRegex = /[\]\}\)]/g
+    // parens only
+    const parsedInput = input.match(parsingRegex)
+    // keep up with pairs 
+    let parenStack = []; 
+
+    // if there are no parens
+    if(parsedInput == null) return true
+    // if we start with an open paren
+    if(parsedInput[0].match(openRegex) == null) return false;
```

Comment:
Considering the simplified version of balanced-parens here too -- how often will these if statements be useful given 1,000 test cases? Overall these do not affect time complexity